src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -5798,6 +5793,9 @@ void ClassFileParser::post_process_parsed_stream(const ClassFileStream* const st
                                                                true,
                                                                CHECK);
     }
+  } else {
+    assert(_class_name == vmSymbols::java_lang_Object(), "already checked");
+    _super_klass = nullptr;
```

Comment:
This seems unnecessary. We have already check the name and `_super_klass` is initialized to null.

Reply:
I would just have:
```
else {
  // The class is Object - so no superclass
}
```

Reply:
My asserts say the same thing, except they comes with proofs that they are not lying.

Reply:
I don't like the 'else' either.